internal/diagnostics/server.go

```diff
@@ -18,6 +19,7 @@ type Server struct {
 	Logger           logr.Logger
 	ProfilingEnabled bool
 	ConfigDumps      util.ConfigDumpDiagnostic
+	ConfigLock       sync.RWMutex
```

Comment:
Where the object is instantiated ? I mean ConfigLock = sync.RWMutex{}

Reply:
Instantiating a Server implicitly instantiates an empty one. Similar to the example in https://tour.golang.org/concurrency/9. Golangisms :shrug: